src/Twig/Extension/SonataAdminExtension.php

```diff
@@ -92,16 +99,26 @@ public function __construct(
                 E_USER_DEPRECATED
             );
         }
+
+        // NEXT_MAJOR: make AdminSubjectExtractor required
+        if (null === $subjectExtractor) {
+            @trigger_error(
+                sprintf('Argument 6 for %s() will be required with the 4.0 release.', __METHOD__),
+                E_USER_DEPRECATED
+            );
+        }
```

Comment:
Or you could have a separate Twig extension just for this, that way there is no need for all the conditionals.

Reply:
It would be fine.